flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/datastream/DataStreamScanRule.scala

```diff
@@ -36,25 +36,32 @@ class DataStreamScanRule
 
   override def matches(call: RelOptRuleCall): Boolean = {
     val scan: FlinkLogicalNativeTableScan = call.rel(0).asInstanceOf[FlinkLogicalNativeTableScan]
-    val dataSetTable = scan.getTable.unwrap(classOf[DataStreamTable[Any]])
-    dataSetTable match {
-      case _: DataStreamTable[Any] =>
-        true
-      case _ =>
-        false
-    }
+    val appendTable = scan.getTable.unwrap(classOf[AppendStreamTable[Any]])
+    val upsertTable = scan.getTable.unwrap(classOf[UpsertStreamTable[Any]])
+
+    appendTable != null || upsertTable != null
```

Comment:
can we avoid those nulls (in multiple places)? For example by using `isInstanceOf` or match? Handling null is dangerous and error prone.

Reply:
I think you are right.